scripts/sommelier

```diff
@@ -67,6 +67,49 @@ init_wine() {
       mv "$WINEPREFIX" "${WINEPREFIX}.bak-$(date +%s)"
   fi
 
+  if [[ -n "${SOMMELIER_INSTALL_MONO:-}" ]]; then
+    if [[ $(grep -c mscoree "$WINEPREFIX"/user.reg 2&>/dev/null) -eq 0 ]]; then
+        WINE_VER=$(strings "$WINE_PLATFORM"/lib/libwine.so.1 | grep "wine-[\.0-9]" | cut -d "-" -f 2)
+        # For future reference MONO_VER https://source.winehq.org/source/dlls/appwiz.cpl/addons.c
+          case "$WINE_VER" in
+            3.0.4)
+              MONO_VER="4.7.1"
+              ;;
+
+            4.0.3)
+              MONO_VER="4.7.5"
+              ;;
+
+            4.21|5.0.3)
+              MONO_VER="4.9.4"
+              ;;&
```

Comment:
What does this `&` do?

Reply:
From `man bash` Using `;;&` in place of `;;` causes the shell to test the next pattern list in the statement, if any, and execute  any associated  list  on a successful match, continuing the case statement execution as if the pattern list had not matched.

Reply:
Aha, thanks! I didn't know that.

So this introduces a bug because of the default condition `*)`. If wine is version `4.21`, it will set `MONO_VER` to `4.9.4`, but keep checking the next cases, eventually ending up on the case `*)`, which changes `MONO_VER` to `6.3.0` and shows the error.

You can just remove the `&` and your code will work correctly. `4.21|5.0.3)` works correctly without `&` because it is a _single condition_ from the point of view of the `case` statement. Even though it checks for two values.

Reply:
As an example: the following outputs `4.9.4`

```shell
WINE_VER="5.0.3"

case "$WINE_VER" in
4.21|5.0.3)
  MONO_VER="4.9.4"
  ;;
*)
  MONO_VER="6.3.0"
  echo "Error can't find wine version! setting wine-mono version to $MONO_VER"
  ;;
esac
```

While the following outputs `6.3.0`:

```shell
WINE_VER="5.0.3"

case "$WINE_VER" in
4.21|5.0.3)
  MONO_VER="4.9.4"
  ;;&
*)
  MONO_VER="6.3.0"
  echo "Error can't find wine version! setting wine-mono version to $MONO_VER"
  ;;
esac
```